### ISSUE TITLE
libfabric with psm3 enabled causing a failed Open MPI library call "fi_endpoint"

### ISSUE THREAD

**Issue by branfosj** [ofiwg/libfabric#6710]
Running various OSU microbenchmarks (5.7) against OpenMPI 4.1.1 I'm getting:
```
--------------------------------------------------------------------------
Open MPI failed an OFI Libfabric library call (fi_endpoint).  This is highly
unusual; your job may behave unpredictably (and/or abort) after this.  Local host: bear-pg0211u17a
  Location: mtl_ofi_component.c:513
  Error: Invalid argument (22)
--------------------------------------------------------------------------
```

This is using libfabric 1.12.1. If I disable psm3 (`--disable-psm3`) then the problem goes away.

I cannot build libfabric 1.12.0 as I hit the `multiple definition of `PACK_SUFFIX'` issue and I do not see the issue when I use libfabric 1.11.2 (as expected, since psm3 was new in 1.12.0).

The difference in output from `fi_info` between a build with and without psm3 disabled is:
```
provider: psm3
    fabric: psm3
    domain: mlx5_0
    version: 112.10
    type: FI_EP_RDM
    protocol: FI_PROTO_PSMX3
provider: psm3
    fabric: psm3
    domain: mlx5_0
    version: 112.10
    type: FI_EP_RDM
    protocol: FI_PROTO_PSMX3
provider: psm3
    fabric: psm3
    domain: mlx5_0
    version: 112.10
    type: FI_EP_RDM
    protocol: FI_PROTO_PSMX3
```

**Comment by branfosj** [comment]
I've put the output of
```
FI_LOG_LEVEL=debug mpirun osu_alltoall -m 16 -x 1000 -i 20000 > ~/debug.out 2>&1
```
at https://gist.github.com/branfosj/432cbb59d631e1035e88ba06145b5e12

**Comment by mshiryaev** [comment]
@branfosj 
>> number of Tx/Rx contexts exceeds limit (1)
if issue is caused by this then try to set env variable PSM3_MULTI_EP=1

**Comment by shefty** [comment]
tagging @acgoldma

**Comment by branfosj** [comment]
Yes, if I set `PSM3_MULTI_EP=1` then the problem goes away.

Investigating further, I see the following in the output of `FI_LOG_LEVEL=debug fi_info`
```
libfabric:531387:psm3:core:psmx3_init_lib():270<info> PSM3 multi-ep feature not available or disabled.
libfabric:531387:psm3:core:psmx3_update_hfi_info():416<info> Tx/Rx contexts: 1 in total, 1 available.
```

But if I run with `PSM3_MULTI_EP=1`, I get this instead
```
libfabric:532331:psm3:core:psmx3_init_lib():268<info> PSM3 multi-ep feature enabled.
libfabric:532331:psm3:core:psmx3_update_hfi_info():416<info> Tx/Rx contexts: 1024 in total, 1024 available.
```

Where I am now confused is in this (from https://ofiwg.github.io/libfabric/master/man/fi_psm3.7.html):
> Scalable endpoints support depends on the multi-EP feature of the PSM3 library. If the PSM3 library supports this feature, the availability is further controlled by an environment variable PSM3_MULTI_EP. **The psm3 provider automatically sets this variable to 1 if it is not set.** The feature can be disabled explicitly by setting PSM3_MULTI_EP to 0.

The bit I emphasised is where the confusion is - I do not see that I am setting this, so I'd expect it to default to 1.

**Comment by acgoldma** [comment]
I believe this may be an issue in a missing patch to the following line: https://github.com/ofiwg/libfabric/blob/e8ec54741adeee156d3e1c018e0fef629488f312/prov/psm3/src/psmx3_init.c#L225 
It looks like we disabled an override to set it to enabled ([link](https://github.com/ofiwg/libfabric/blob/e8ec54741adeee156d3e1c018e0fef629488f312/prov/psm3/src/psmx3_init.c#L247)), then forgot the 2nd part of the patch:
```
@@ -51,7 +51,7 @@ struct psmx3_env psmx3_env = {
        .conn_timeout   = 10,
        .prog_interval  = -1,
        .prog_affinity  = NULL,
-       .multi_ep       = 0,
+       .multi_ep       = 1,
        .inject_size    = 64,
        .lock_level     = 2,
        .lazy_conn      = 0,
@@ -222,7 +222,7 @@ static int psmx3_check_multi_ep_cap(void)
        uint64_t caps = PSM2_MULTI_EP_CAP;
        char *s = getenv("PSM3_MULTI_EP");

-       if (psm2_get_capability_mask(caps) == caps && psmx3_get_yes_no(s, 0))
+       if (psm2_get_capability_mask(caps) == caps && psmx3_get_yes_no(s, 1))
                psmx3_env.multi_ep = 1;
        else
                psmx3_env.multi_ep = 0;
```

We caught this internally, but I was delayed pushing the fix upstream. I will try to push it A.S.A.P.

**Comment by branfosj** [comment]
Thanks. I can confirm that that fixes the issue for me.

**Comment by acgoldma** [comment]
This should be fixed with #6877
